UI/scripts/script.js

```diff
@@ -117,54 +117,71 @@ sendCommandBtn.addEventListener('click', function() {
         }
     };
 
-    const request = new XMLHttpRequest();
-    const url = 'http://localhost:8080/settelemetry?id=1'
-    request.open("PUT", url);
-    request.send(JSON.stringify(formData));
-
-     // Send alert if request is successful
-     request.onreadystatechange = function () {
-        if (this.readyState == 4) {
-            if (this.status == 200) {
-                window.alert('Request to set telemetry has been sent.');
-            } else {
-                window.alert('Failed to send telemetry request. Status: ' + this.status);
+    if (Number(X) < -180 || Number(X) > 360 || Number(Y) < -180 || Number(Y) > 360 || Number(Z) < -180 || Number(Z) > 360 ||
+    Number(pitch) < -180 || Number(pitch) > 360 || Number(yaw) < -180 || Number(yaw) > 360 || Number(roll) < -180 || Number(roll) > 360 ||
+    X == "" || Y == "" || Z == "" || pitch == "" || yaw == "" || roll == "")
+    {
+        window.alert('All input fields must be between -180 and 360.')
+    }
+    else
+    {
+        const request = new XMLHttpRequest();
+        const url = 'http://localhost:8080/settelemetry?id=1'
+        request.open("PUT", url);
+        request.send(JSON.stringify(formData));
+    
+         // Send alert if request is successful
+         request.onreadystatechange = function () {
+            if (this.readyState == 4) {
+                if (this.status == 200) {
+                    window.alert('Request to set telemetry has been sent.');
+                } else {
+                    window.alert('Failed to send telemetry request. Status: ' + this.status);
+                }
             }
-        }
-    };
+        };
+    }
 });
 
 // Event listener for requesting coordinates to Space CNDH
 sendCommandBtn2.addEventListener('click', function() {
```

Comment:
I did try this strategy originally, however I was getting errors regarding multiple buttons being named the same thing. This lead me to move to each button with a different name and function. If I were better at JS or HTML I'm sure it could be done but alas.

Reply:
Ah, fair enough o7

Reply:
Just noticed the code is all unique, ifgnore this.